File: abc161/b/b.py
def vote(n, m, a):
    check = 0
    total = sum(a)
    for i in a:
        if i * 4 * m >= total:
            check += 1
            if check == m:
                return True
    return False


def main():
    n, m = map(int, input().split())
    a = list(map(int, input().split()))
    if vote(n, m, a):
        print("Yes")
    else:
        print("No")


if __name__ == "__main__":
    main()

File: abc161/b/b.ts
import * as fs from "fs";

function vote(n: number, m: number, a: number[], total: number): boolean {
  let check: number = 0;
  for (let i = 0; i < n; i++) {
    if (a[i] * 4 * m >= total) {
      check++;
      if (check == m) {
        return true;
      }
    }
  }
  return false;
}

function main() {
  const input = fs.readFileSync("/dev/stdin", "utf-8").trim().split(/\s+/).map(Number);
  const n = input[0];
  const m = input[1];
  const a = input.slice(2);
  let total: number = 0;
  for (let i = 0; i < a.length; i++) {
    total += a[i];
  }
  if (vote(n, m, a, total)) {
    console.log("Yes");
  }
  else {
    console.log("No")
  }
}

main();
